gson/src/main/java/com/google/gson/internal/bind/ReflectiveTypeAdapterFactory.java

```diff
@@ -126,10 +126,10 @@ private ReflectiveTypeAdapterFactory.BoundField createBoundField(
       final TypeToken<?> fieldType, boolean serialize, boolean deserialize,
       final boolean blockInaccessible) {
     final boolean isPrimitive = Primitives.isPrimitive(fieldType.getRawType());
-    // special casing primitives here saves ~5% on Android...
```

Comment:
Have removed this comment because it is a bit misleading and even when it was [added in 2011](https://github.com/google/gson/commit/4b042671af018f308c2a261145dc6712e39b96e7#diff-98964cf2f1728356669a5e22fba88f24d52a20c6fc974f36c798976cb3fc0ee4R131) it is not really clear to me what exactly this is about, or which special casing was used to check this performance impact.